doc/nrf/security/ap_protect.rst

```diff
@@ -197,16 +216,98 @@ This is the default setting in the |NCS|.
 
 You can start debugging the firmware without additional steps needed.
 
-Once you are done debugging, run the following command to enable the access port protection:
+.. _app_approtect_uicr_approtect:
+
+Enabling AP-Protect by locking the UICR.APPROTECT register
+==========================================================
+
+For the devices that are in a production environment, you are highly recommended to lock the UICR.APPROTECT register to prevent unauthorized access to the device.
+If the access port protection is configured this way, it cannot be disabled without erasing the flash memory.
+
+.. note::
+    This is the only mechanism supported for the nRF52 Series and the nRF9160 devices that do not support hardware and software for AP-Protect.
+
+To lock the UICR.APPROTECT register, complete the following steps:
 
 .. code-block:: console
 
    nrfjprog --rbp ALL
 
-This command enables the AP-Protect and resets the device.
+This command enables the AP-Protect (and Secure AP-Protect) and resets the device.
+
+.. _app_secure_approtect:
+
+Secure AP-Protect
+=================
+
+With Trusted Firmware-M (TF-M) comes :ref:`security by separation <app_boards_spe_nspe>`, enabling a Secure Processing Environment (SPE) that is isolated from the Non-Secure Processing Environment (NSPE).
+TF-M is available for the nRF53 and nRF91 Series devices.
+
+While AP-Protect blocks access to all CPU registers and memories, Secure AP-Protect limits the CPU access only for the non-secure accesses.
+This allows debugging of the NSPE, while the SPE debugging is blocked.
+
+The following Kconfig options for enabling Secure AP-Protect are available for the nRF91x1 and nRF53 Series devices:
+
+* :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_LOCK`
+* :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_USER_HANDLING`
+* :kconfig:option:`CONFIG_NRF_APPROTECT_USE_UICR`
+
+In addition, you can enable Secure AP-Protect by setting the UICR.SECUREAPPROTECT register.
+
+Enabling Secure AP-Protect with :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_LOCK`
+----------------------------------------------------------------------------------
+
+Setting the :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_LOCK` Kconfig option to ``y`` and compiling the firmware enables the secure access protection mechanism for SoCs of the nRF53 Series.
```

Comment:
How about the 91? This Kconfig options doesn't depend on the SOC being a 53.

Reply:
Decided to skip that due to the issue illustrated in important-section.